docs/en/migrations/bigquery.md

```diff
@@ -0,0 +1,110 @@
+
+| Table                                                                                             | Rows          | Files Exported | Data Size | BigQuery Export | Slot Time       | ClickHouse Import |
+| ------------------------------------------------------------------------------------------------- | ------------- | -------------- | --------- | --------------- | --------------- | ----------------- |
+| [blocks](https://github.com/ClickHouse/examples/blob/main/ethereum/schemas/blocks.md)             | 16,569,489    | 73             | 14.53GB   | 23 secs         | 37 min          | 15.4 secs         |
+| [transactions](https://github.com/ClickHouse/examples/blob/main/ethereum/schemas/transactions.md) | 1,864,514,414 | 5169           | 957GB     | 1 min 38 sec    | 1 day 8hrs      | 18 mins 5 secs    |
+| [traces](https://github.com/ClickHouse/examples/blob/main/ethereum/schemas/traces.md)             | 6,325,819,306 | 17,985         | 2.896TB   | 5 min 46 sec    | 5 days 19 hr    | 34 mins 55 secs   |
+| [contracts](https://github.com/ClickHouse/examples/blob/main/ethereum/schemas/contracts.md)       | 57,225,837    | 350            | 45.35GB   | 16 sec          | 1 hr 51 min     | 39.4 secs         |
+| Total                                                                                             | 8.26 billion  | 23,577         | 3.982TB   | 8 min 3 sec     | \> 6 days 5 hrs | 53 mins 45 secs   |
+
+## 1. Export table data to GCS
+
+In this step, we utilize the [BigQuery SQL workspace](https://cloud.google.com/bigquery/docs/bigquery-web-ui) to execute our SQL commands. Below, we export a BigQuery table named `mytable` to a GCS bucket using the [`EXPORT DATA`](https://cloud.google.com/bigquery/docs/reference/standard-sql/other-statements) statement.
+
+```sql
+EXPORT DATA
+    OPTIONS (
+        uri = 'gs://mybucket/mytable/*.parquet',
+        format = 'PARQUET',
+        overwrite = true
+    )
+AS (
+    SELECT * FROM mytable
+);
+```
+
+In the above query, we export our BigQuery table to the [Parquet data format](https://parquet.apache.org/). We also have a `*` character in our `uri` parameter. This ensures the output is sharded into multiple files, with a numerically increasing suffix, should the export exceed 1GB of data.
+
+This approach has a number of advantages:
+
+- Google allows up to 50TB per day to be exported to GCS for free. Users only pay for GCS storage.
+- Exports produce multiple files automatically, limiting each to a maximum of 1GB of table data. This is beneficial to ClickHouse since it allows imports to be parallelized.
+- Parquet, as a column-oriented format, represents a better interchange format since it is inherently compressed and faster for BigQuery to export and ClickHouse to query
```

Comment:
For larger exports the following can be used:

```
DECLARE export_path string;
DECLARE n int64;
DECLARE i INT64;
SET i = 0;
SET n = 100;


WHILE i < n DO
  SET export_path = CONCAT('gs://mybucket/mytable/', i,'-*.parquet');
  EXPORT DATA
    OPTIONS (
    uri = (export_path),
    format = 'PARQUET',
    overwrite = true
    )
  AS (
    SELECT *
      FROM mytable
    WHERE export_id = i
  );
  SET i = i + 1;
END WHILE;
```

Reply:
In this query, is there an optimal `n` we should suggest, like if 1TB split it up into `n` files per `x` GB?

Also, can we use this query even for small datasets? I'd like to simplify the guide so just one query for both small and large imports would be easier in my opinion for the user.

Reply:
this is really when you have over 5 billion rows. Maybe total rows/5billion = N